midend/lib/Utils/DIPUtils.cpp

```diff
@@ -410,6 +411,185 @@ Value customTanVal(OpBuilder &builder, Location loc, Value angleVal) {
+  affineMatrix[2] = builder.create<arith::NegFOp>(loc, negB1);
+  affineMatrix[5] = builder.create<arith::NegFOp>(loc, negB2);
+}
+
+void affineTransformController(OpBuilder &builder, Location loc,
+                               MLIRContext *ctx, Value input, Value output,
+                               SmallVector<Value, 6> affineMatrix,
+                               int64_t stride) {
+  VectorType vectorTyF32 = VectorType::get({stride}, FloatType::getF32(ctx));
+
+  Value c0 = builder.create<arith::ConstantIndexOp>(loc, 0);
+  Value c1 = builder.create<arith::ConstantIndexOp>(loc, 1);
+  Value c512F32 =
+      builder.create<arith::ConstantOp>(loc, builder.getF32FloatAttr(512));
+  Value c1024F32 =
+      builder.create<arith::ConstantOp>(loc, builder.getF32FloatAttr(1024));
+  Value c512F32Vec = builder.create<vector::SplatOp>(loc, vectorTyF32, c512F32);
+  Value c1024F32Vec =
+      builder.create<vector::SplatOp>(loc, vectorTyF32, c1024F32);
+
+  inverseAffineMatrix(builder, loc, affineMatrix);
+
+  Value m0Vec =
+      builder.create<vector::SplatOp>(loc, vectorTyF32, affineMatrix[0]);
+  Value m1Vec =
+      builder.create<vector::SplatOp>(loc, vectorTyF32, affineMatrix[1]);
+  Value m2Vec =
+      builder.create<vector::SplatOp>(loc, vectorTyF32, affineMatrix[2]);
+  Value m3Vec =
+      builder.create<vector::SplatOp>(loc, vectorTyF32, affineMatrix[3]);
+  Value m4Vec =
+      builder.create<vector::SplatOp>(loc, vectorTyF32, affineMatrix[4]);
+  Value m5Vec =
+      builder.create<vector::SplatOp>(loc, vectorTyF32, affineMatrix[5]);
+
+  Value inputRow = builder.create<memref::DimOp>(loc, input, c0);
+  Value inputCol = builder.create<memref::DimOp>(loc, input, c1);
+
+  Value outputRow = builder.create<memref::DimOp>(loc, output, c0);
+  Value outputCol = builder.create<memref::DimOp>(loc, output, c1);
+
+  Value strideVal = builder.create<arith::ConstantIndexOp>(loc, stride);
+
+  // generate vector [0., 1., ..., stride - 1]
+  std::vector<float> xInit(stride);
+  std::iota(xInit.begin(), xInit.end(), .0);
+  Value xVecInitial = builder.create<arith::ConstantOp>(
+      loc,
+      DenseFPElementsAttr::get(VectorType::get(stride, FloatType::getF32(ctx)),
+                               ArrayRef<float>(xInit)));
+
+  builder.create<scf::ParallelOp>(
+      loc,
+      /*lowerBounds=*/ValueRange{c0, c0},
+      /*upperBounds=*/ValueRange{outputRow, outputCol},
+      /*steps=*/ValueRange{c1, strideVal},
+      /*bodyBuilderFn=*/
+      [&](OpBuilder &builderFn, Location locFn, ValueRange ivsFn) {
+        Value yVec = builderFn.create<vector::SplatOp>(
+            locFn, vectorTyF32, indexToF32(builderFn, locFn, ivsFn[0]));
+        Value delta = builderFn.create<vector::SplatOp>(locFn, vectorTyF32, indexToF32(builderFn, locFn, ivsFn[1]));
+        Value xVec = builderFn.create<arith::AddFOp>(locFn, xVecInitial, delta);
+
+        // sourceXVec=(xVec*m0Vec*1024+(yVec*m1Vec+m2Vec)*1024+512)/1024
```

Comment:
Please replace 1024 and 512 with appropriate image dimension variables, since this is not the general case and we intend to work with images of different dimensions.

Reply:
Put these into configuration?

Reply:
> Put these into configuration?

Can you elaborate.

Reply:
1024 and 512 are not dimension-based parameters. It depends on how much precision you need when converting a float to an int. Of course, it seems to have other factors to consider that also come into play in Opencv's remap function. I have plans for the next week to put together the full flow of Opencv rotations.

Reply:
Ahh, I see